src/main/java/uk/gov/register/store/postgres/PostgresDriverTransactional.java

```diff
@@ -38,18 +39,21 @@ private PostgresDriverTransactional(Handle handle, TransactionalMemoizationStore
         this.stagedEntries = new ArrayList<>();
         this.stagedItems = new HashMap<>();
         this.stagedCurrentKeys = new HashMap<>();
+        this.initialEntryCount = super.getTotalEntries();
```

Comment:
`initialEntryCount` also needs to be reset whenever we do `writeEntries()` (after the `isEmpty()` check). Currently, if we flush the staged entries half way through a transaction (e.g. if there is an assert-root-hash half way through the file) then we will return 0 from `getTotalEntries()` even though there are entries in the database.

Maybe we should call it something like `committedEntryCount` since it can change after being initialised?